server.js

```diff
@@ -23,43 +24,39 @@ const port = process.env.PORT || 3000
 
 app.prepare().then(() => {
   createServer((req, res) => {
-    const parsedUrl = parse(req.url, true)
-    const { pathname, query } = parsedUrl
+    let parsedUrl = parseURL(req.url)
+    const { pathname, query: queryStr } = parsedUrl
     const host = req.headers.host
 
-    /*
-     * HTTP redirects
-     */
     let [redirectCode, redirectLocation] = getRedirect(host, pathname, {
       req,
       dev
     })
-
     if (redirectLocation) {
-      // should be getting the query as a string
-      const { query } = parse(req.url)
-      if (query) {
-        redirectLocation += '?' + query
+      // HTTP redirects
+
+      if (queryStr) {
+        redirectLocation += '?' + queryStr
       }
       res.writeHead(redirectCode, {
         'Cache-control': 'no-cache',
         Location: redirectLocation
       })
       res.end()
     } else if (/^\/doc(\/.*)?$/.test(pathname)) {
-      /*
-       * Docs Engine handler
-       */
+      // Docs Engine handler
 
       // Force 404 response code for any inexistent /doc item.
       if (!getItemByPath(pathname)) {
         res.statusCode = 404
       }
 
       // Custom route for all docs
-      app.render(req, res, '/doc', query)
+      app.render(req, res, '/doc', parseQuery(queryStr))
     } else {
       // Regular Next.js handler
+
+      if (null === parsedUrl.query) parsedUrl.query = {}
```

Comment:
does it worth doing this minor optimization considering that we have to this hack?

Reply:
Probably not.

I don't see it as a hack though, just part of the necessary custom Next server logic.

But anyway should I just revert this whole file and leave the redirect file changes (totally cosmetic)? If this whole PR is pointless at this point, please just close it 🙂

Reply:
up to you @jorgeorpinel  :)

Reply:
OK I reverted the optimization and hack. But then I found a better one! Implemented in 08522d88, please check.